### PR TITLE
Enable mypy for langgraph lib

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: libs/checkpoint/langgraph/checkpoint/serde/types.py [hunk 1]
```diff
@@ -31,13 +28,7 @@ def UpdateType(self) -> Any: ...
 
     def checkpoint(self) -> Optional[C]: ...
 
-    def from_checkpoint(
-        self, checkpoint: Optional[C], config: RunnableConfig
-    ) -> Generator[Self, None, None]: ...
-
-    async def afrom_checkpoint(
```
Comment (hinthornw) [inline libs/checkpoint/langgraph/checkpoint/serde/types.py:38]: removing async?

File: libs/langgraph/langgraph/channels/named_barrier_value.py [hunk 2]
```diff
@@ -11,10 +11,13 @@ class NamedBarrierValue(Generic[Value], BaseChannel[Value, Value, set[Value]]):
...
     __slots__ = ("names", "seen")
 
+    names: set[Value]
+    seen: set[Value]
+
     def __init__(self, typ: Type[Value], names: set[Value]) -> None:
         super().__init__(typ)
         self.names = names
-        self.seen = set()
+        self.seen: set[str] = set()
```
Comment (hinthornw) [inline libs/langgraph/langgraph/channels/named_barrier_value.py:20]: Why type as `str` rather than `Value` ?

File: libs/langgraph/langgraph/graph/graph.py [hunk 3]
```diff
@@ -178,14 +188,20 @@ def add_node(
...
         if not isinstance(node, str):
             action = node
-            node = getattr(action, "name", action.__name__)
+            node = getattr(action, "name", getattr(action, "__name__"))
+            if node is None:
+                raise ValueError(
+                    "Node name must be provided if action is not a function"
+                )
+        if action is None:
+            raise RuntimeError
```
Comment (hinthornw) [inline libs/langgraph/langgraph/graph/graph.py:197]:
```suggestion
            raise RuntimeError(f"Expected a function or Runnable action in add_node. Received None.")
````

File: libs/langgraph/langgraph/graph/state.py [hunk 4]
```diff
@@ -318,14 +320,20 @@ def add_node(
             )
         if not isinstance(node, str):
             action = node
-            node = getattr(action, "name", action.__name__)
+            node = getattr(action, "name", getattr(action, "__name__"))
```
Comment (hinthornw) [inline libs/langgraph/langgraph/graph/state.py:323]:
```suggestion
            node = getattr(action, "name", getattr(action, "__name__", None))
```

File: libs/langgraph/langgraph/graph/state.py [hunk 5]
```diff
@@ -702,11 +714,23 @@ def _get_channels(
         if name != "__slots__"
     }
     return (
-        {k: v for k, v in all_keys.items() if not is_managed_value(v)},
+        {k: v for k, v in all_keys.items() if isinstance(v, BaseChannel)},
```
Comment (hinthornw) [inline libs/langgraph/langgraph/graph/state.py:717]: Why this change?

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py [hunk 6]
```diff
@@ -129,15 +130,15 @@ def _get_model_preprocessing_runnable(
...
+    model: BaseChatModel,
     tools: Union[ToolExecutor, Sequence[BaseTool], ToolNode],
     *,
     state_schema: Optional[StateSchemaType] = None,
     messages_modifier: Optional[MessagesModifier] = None,
     state_modifier: Optional[StateModifier] = None,
     checkpointer: Optional[BaseCheckpointSaver] = None,
-    interrupt_before: Optional[Sequence[str]] = None,
-    interrupt_after: Optional[Sequence[str]] = None,
+    interrupt_before: Optional[list[str]] = None,
```
Comment (hinthornw) [inline libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:140]: ooc - why the change here?

File: libs/langgraph/langgraph/utils/queue.py [hunk 7]
```diff
@@ -1,10 +1,13 @@
+# type: ignore
```
Comment (hinthornw) [inline libs/langgraph/langgraph/utils/queue.py:1]: nice

**Review by hinthornw — APPROVED** [review]

File: libs/checkpoint/langgraph/checkpoint/serde/types.py (hunk 1 above)
Comment (nfcampos) [reply libs/checkpoint/langgraph/checkpoint/serde/types.py:38]: This was removed from the actual lib a while ago, but had been left in this protocol in checkpoint lib

File: libs/langgraph/langgraph/graph/state.py (hunk 5 above)
Comment (nfcampos) [reply libs/langgraph/langgraph/graph/state.py:717]: Because mypy doesn't know that everything that isn't a managed value is a base channel

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py (hunk 6 above)
Comment (nfcampos) [reply libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:140]: Because the type accepted by compile changed too
